File: demo-vk-rest-scraping.java/src/main/java/demo/reader/Reader.java
package demo.reader;

import demo.http.HttpRequester;

public class Reader<T> {

    private static final String URL_PREFIX = "https://api.vk.com/method/database.";
    private static final String URL_SUFFIX = "&lang=en&count=10000";

    private final HttpRequester<T> httpRequester;

    public Reader(Class<T> clazz) {
        this.httpRequester = new HttpRequester<>(clazz);
    }

    public T read(String urlBody) {
        return httpRequester.request(URL_PREFIX + urlBody + URL_SUFFIX);
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/reader/Reader.kt
package demo.reader

import demo.http.HttpRequester

open class Reader<T>(clazz: Class<T>) {

    private val httpRequester: HttpRequester<T>

    init {
        this.httpRequester = HttpRequester(clazz)
    }

    open fun read(urlBody: String): T {
        return httpRequester.request(URL_PREFIX + urlBody + URL_SUFFIX)
    }

    companion object {

        private val URL_PREFIX = "https://api.vk.com/method/database."
        private val URL_SUFFIX = "&lang=en&count=10000"
    }
}
